aya/src/sys/bpf.rs

```diff
@@ -93,9 +93,9 @@ pub(crate) fn bpf_create_map(
     // The map name was added as a parameter in kernel 4.15+ so we skip adding it on
     // older kernels for compatibility
     if KernelVersion::at_least(4, 15, 0) {
+        let name_bytes = name.to_bytes();
         // u.map_name is 16 bytes max and must be NULL terminated
-        let name_bytes = name.to_bytes_with_nul();
-        let len = cmp::min(name_bytes.len(), u.map_name.len());
+        let len = cmp::min(name_bytes.len(), u.map_name.len() - 1);
```

Comment:
Hmmm I annoyed at myself:
https://github.com/aya-rs/aya/pull/1165#discussion_r1941792631

I should have caught that using `cmp::min` would strip the nul byte 😭 

Do you recall offhand if the rules are the same for program names as the logic there was changed in the same commit

Reply:
Yes, exactly the same rule applies to program names, but that was already fixed in 9eefb48a0ad90831c09c7bd941d035b34e1f1771:

https://github.com/aya-rs/aya/blob/dcf66db845bc9b2d088dbd4096a279f346abb0c2/aya/src/sys/bpf.rs#L153